File: bindings/lua/Polycode/Label.lua
require "Polycode/Image"

class "Label" (Image)



Label.ANTIALIAS_FULL = 0
Label.ANTIALIAS_NONE = 1
Label.ANTIALIAS_STRONG = 2
Label.ANTIALIAS_LCD = 3
Label.ANTIALIAS_LCD_HINT = 4
Label.ANTIALIAS_FULL_HINT = 5




function Label:Label(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "Image" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycore.Label(unpack(arg))
	end
end

function Label:setText(text)
	local retVal = Polycore.Label_setText(self.__ptr, text)
end

function Label:getText()
	local retVal =  Polycore.Label_getText(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["String"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Label:getTextWidthForString(text)
	local retVal = Polycore.Label_getTextWidthForString(self.__ptr, text)
	return retVal
end

function Label:getTextHeightForString(text)
	local retVal = Polycore.Label_getTextHeightForString(self.__ptr, text)
	return retVal
end

function Label:getTextWidth()
	local retVal =  Polycore.Label_getTextWidth(self.__ptr)
	return retVal
end

function Label:getTextHeight()
	local retVal =  Polycore.Label_getTextHeight(self.__ptr)
	return retVal
end

function Label:setColorForRange(color, rangeStart, rangeEnd)
	local retVal = Polycore.Label_setColorForRange(self.__ptr, color.__ptr, rangeStart, rangeEnd)
end

function Label:clearColors()
	local retVal =  Polycore.Label_clearColors(self.__ptr)
end

function Label:getColorForIndex(index)
	local retVal = Polycore.Label_getColorForIndex(self.__ptr, index)
	if retVal == nil then return nil end
	local __c = _G["Color"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Label:getPremultiplyAlpha()
	local retVal =  Polycore.Label_getPremultiplyAlpha(self.__ptr)
	return retVal
end

function Label:setPremultiplyAlpha(val)
	local retVal = Polycore.Label_setPremultiplyAlpha(self.__ptr, val)
end

function Label:setFont(newFont)
	local retVal = Polycore.Label_setFont(self.__ptr, newFont.__ptr)
end

function Label:getFont()
	local retVal =  Polycore.Label_getFont(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Font"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Label:setSize(newSize)
	local retVal = Polycore.Label_setSize(self.__ptr, newSize)
end

function Label:getSize()
	local retVal =  Polycore.Label_getSize(self.__ptr)
	return retVal
end

function Label:getAntialiasMode()
	local retVal =  Polycore.Label_getAntialiasMode(self.__ptr)
	return retVal
end

function Label:setAntialiasMode(newMode)
	local retVal = Polycore.Label_setAntialiasMode(self.__ptr, newMode)
end

function Label:getBaselineAdjust()
	local retVal =  Polycore.Label_getBaselineAdjust(self.__ptr)
	return retVal
end

function Label:setBackgroundColor(color)
	local retVal = Polycore.Label_setBackgroundColor(self.__ptr, color.__ptr)
end

function Label:setForegroundColor(color)
	local retVal = Polycore.Label_setForegroundColor(self.__ptr, color.__ptr)
end

function Label:setColors(backgroundColor, foregroundColor)
	local retVal = Polycore.Label_setColors(self.__ptr, backgroundColor.__ptr, foregroundColor.__ptr)
end

function Label:getBackgroundColor()
	local retVal =  Polycore.Label_getBackgroundColor(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Color"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Label:getForegroundColor()
	local retVal =  Polycore.Label_getForegroundColor(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Color"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Label:optionsChanged()
	local retVal =  Polycore.Label_optionsChanged(self.__ptr)
	return retVal
end

function Label:__delete()
	if self then Polycore.delete_Label(self.__ptr) end
end

File: bindings/javascript/Polycode/Label.js
require('Polycode/Image')

function Label(font,text,size,antiAliasMode,premultiplyAlpha,backgroundColor,foregroundColor) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.Label(font,text,size,antiAliasMode,premultiplyAlpha,backgroundColor,foregroundColor)
	}
}

Label.ANTIALIAS_FULL = 0
Label.ANTIALIAS_NONE = 1
Label.ANTIALIAS_STRONG = 2
Label.ANTIALIAS_LCD = 3
Label.ANTIALIAS_LCD_HINT = 4
Label.ANTIALIAS_FULL_HINT = 5

Label.prototype = Object.create(Image.prototype)

Duktape.fin(Label.prototype, function (x) {
	if (x === Label.prototype) {
		return;
	}
	Polycode.Label__delete(x.__ptr)
})

Label.prototype.setText = function(text) {
	Polycode.Label_setText(this.__ptr, text)
}

Label.prototype.getText = function() {
	return Polycode.Label_getText(this.__ptr)
}

Label.prototype.getTextWidthForString = function(text) {
	return Polycode.Label_getTextWidthForString(this.__ptr, text)
}

Label.prototype.getTextHeightForString = function(text) {
	return Polycode.Label_getTextHeightForString(this.__ptr, text)
}

Label.prototype.getTextWidth = function() {
	return Polycode.Label_getTextWidth(this.__ptr)
}

Label.prototype.getTextHeight = function() {
	return Polycode.Label_getTextHeight(this.__ptr)
}

Label.prototype.setColorForRange = function(color,rangeStart,rangeEnd) {
	Polycode.Label_setColorForRange(this.__ptr, color, rangeStart, rangeEnd)
}

Label.prototype.clearColors = function() {
	Polycode.Label_clearColors(this.__ptr)
}

Label.prototype.getColorForIndex = function(index) {
	var retVal = new Color()
	retVal.__ptr = Polycode.Label_getColorForIndex(this.__ptr, index)
	return retVal
}

Label.prototype.getPremultiplyAlpha = function() {
	return Polycode.Label_getPremultiplyAlpha(this.__ptr)
}

Label.prototype.setPremultiplyAlpha = function(val) {
	Polycode.Label_setPremultiplyAlpha(this.__ptr, val)
}

Label.prototype.setFont = function(newFont) {
	Polycode.Label_setFont(this.__ptr, newFont)
}

Label.prototype.getFont = function() {
	var retVal = new Font()
	retVal.__ptr = Polycode.Label_getFont(this.__ptr)
	return retVal
}

Label.prototype.setSize = function(newSize) {
	Polycode.Label_setSize(this.__ptr, newSize)
}

Label.prototype.getSize = function() {
	return Polycode.Label_getSize(this.__ptr)
}

Label.prototype.getAntialiasMode = function() {
	return Polycode.Label_getAntialiasMode(this.__ptr)
}

Label.prototype.setAntialiasMode = function(newMode) {
	Polycode.Label_setAntialiasMode(this.__ptr, newMode)
}

Label.prototype.getBaselineAdjust = function() {
	return Polycode.Label_getBaselineAdjust(this.__ptr)
}

Label.prototype.setBackgroundColor = function(color) {
	Polycode.Label_setBackgroundColor(this.__ptr, color)
}

Label.prototype.setForegroundColor = function(color) {
	Polycode.Label_setForegroundColor(this.__ptr, color)
}

Label.prototype.setColors = function(backgroundColor,foregroundColor) {
	Polycode.Label_setColors(this.__ptr, backgroundColor, foregroundColor)
}

Label.prototype.getBackgroundColor = function() {
	var retVal = new Color()
	retVal.__ptr = Polycode.Label_getBackgroundColor(this.__ptr)
	return retVal
}

Label.prototype.getForegroundColor = function() {
	var retVal = new Color()
	retVal.__ptr = Polycode.Label_getForegroundColor(this.__ptr)
	return retVal
}

Label.prototype.optionsChanged = function() {
	return Polycode.Label_optionsChanged(this.__ptr)
}
